File: server/Content/ContentManager.js
class ContentManager {
  /**
   * Manages content and makes API calls to the database
   * Fetches data from the database
   */
  constructor(user) {
    this.username = user.username;
    this.password = user.password;
  }

  accessDatabase() {
    // Implementation for database access
  }

  makeFlashdeck() {
    // Implementation for creating a flash deck
  }

  makeFlashcards() {
    const front = "f1";
    const back = "b1";
    const card = new Flashcard(front, back);
  }

  addCardToDeck() {
    // Implementation for adding a card to a deck
  }

  deleteCardFromDeck() {
    // Implementation for deleting a card from a deck
  }

  practiceFlashdeck(flashdeck) {
    // Implementation for practicing a flash deck
    // Iterate through the flash deck
    // Maintain separate lists for known and unknown cards
  }
}

File: server/Content/ContentManager.py
from mongoDB.RESTful_api import Database_Manager
import Flashcard

class ContentManager:
    """
    Manages content and makes api calls to the database
    fetches data from the database
    """

    def __init__(self, user):
        user.username
        user.password

    def accessDatabase(self):
        pass

    def makeflashdeck(self):
        pass

    def makeflashcards(self):
        front = "f1"
        back = "b1"
        card = Flashcard(front, back)

    def addcardtodeck(self):
        pass
    
    def deletecardfromdeck(self):
        pass

    def practiceflashdeck(self, flashdeck): #Here or interactions manager?
        #get flashdeck
        #iterate flashdeck
        #have another 2 list for cards they know and cards they don't know?
        pass